app/controllers/users_controller.rb

```diff
@@ -1,8 +1,91 @@
 class UsersController < ApplicationController
-  def index; end
+  def index
+    @logged_user = current_user
+  end
 
   def show
+    @logged_user = current_user
     @user = User.find(params[:id])
     @recent_posts = @user.recent_posts
   end
+
+  def new
+    @comment = Comment.new
+    @like = Like.new
+    @post = Post.new
+  end
+
+  def create
+    begin
+      params.require(:post)
+    rescue StandardError
+      # Do nothing
+    else
+      handle_post_creation
+    end
+
+    begin
+      params.require(:comment)
+    rescue StandardError
+      # Do nothing
+    else
+      handle_comment_creation
+    end
+
+    begin
+      params.require(:like)
+    rescue StandardError
+      # Do nothing
+    else
+      handle_like_creation
+    end
+  end
+
+  private
+
+  def handle_post_creation
+    @post = Post.new(params.require(:post).permit(:author_id, :title, :text, :text, :comments_counter, :likes_counter))
+
+    respond_to do |format|
+      format.html do
+        if @post.save
+          flash[:success] = 'Question saved successfully'
+          redirect_to user_posts_path(@post.user.id)
+        else
+          flash.now[:error] = 'Error: Question could not be saved'
+        end
+      end
+    end
+  end
+
+  def handle_comment_creation
+    @logged_user = current_user
+    @comment = Comment.new(params.require(:comment).permit(:author_id, :post_id, :text))
+
+    respond_to do |format|
+      format.html do
+        if @comment.save
+          flash[:success] = 'Question saved successfully'
+          redirect_to request.referrer
+        else
+          flash.now[:error] = 'Error: Question could not be saved'
+        end
+      end
+    end
+  end
+
+  def handle_like_creation
+    @like = Like.new(params.require(:like).permit(:author_id, :post_id))
+
+    respond_to do |format|
+      format.html do
+        if @like.save
+          flash[:success] = 'Question saved successfully'
+          redirect_to request.referrer
+        else
+          flash.now[:error] = 'Error: Question could not be saved'
+        end
+      end
+    end
+  end
```

Comment:
- [ ] It would be nice if we used the rails' conventional way of working with controllers. Every controller action i.e. **`index`**, **`show`**, **`create`**, **`new`**, **`update`** and **`delete`**, should only interact with the associated models.  

   With that in mind, **`Comment`**, **`Like`**, and **`Post`** can only be created in their own separate controllers.
   
   This approach has also led to not keeping our code DRY(Don't Repeat Yourself). In the **`PostsController`**, all the logic implemented in this **`UsersController`** have been repeated, which is not acceptable for it is not a good code practice.

   Please consider generating and implementing logic for the following controllers:

1.

```ruby
# controllers/comments_controller.rb

class CommentsController < ApplicationController
  
  def new
    # add logic here for creating comment
  end

  def create
    # add logic for creating a comment
  end

  # remember to use strong parameters as demonstrated below

  private 

  def comment_params
    params.require(:comment).permit(:text)
  end


end
```

2.

```ruby
# controllers/likes_controller.rb

  class LikesController < ApplicationController
    def create
      # add logic for creating a like here
    end
  end
```

### NOTE: Since you are going to create these two Controllers, remember to reconfigure your routes to work with them correctly.

Reply:
If I do that I'm repeating the week, that's good news 👍 . I want to know exactly where in the lesson we've been taught these conventions for controllers. Maybe it was here: https://tatyanacodes.wordpress.com/2015/08/17/build-a-rails-form-and-skip-the-gems-part-ii-all-about-the-form/? Creating the new action in a different controller in the material for this lesson... Worst than that is that we don't see anymore a lot of websites with dedicated pages for getting user input... but I'll do my best to do those changes :)

Reply:
## IMPORTANT ## 

I think It's impossible to configure the routes to work properly in all cases, I'm always creating 2 models at least per view 👍 , if someone explain to me how to do it in my project I'll do that changes